src/app/use-cases/request-signature.ts

```diff
@@ -28,6 +36,27 @@ export type RequestSignaturePayload = {
   productId: Product["id"];
 };
 
+export type RequestSignatureStatusPayload = {
+  subscriptionId: Subscription["id"];
+  signatureRequestId: SignatureRequest["id"];
+  signatureRequestStatus: SignatureRequest["status"];
+};
+
+const isReadyStatus = (u: unknown): u is SignatureRequestStatus =>
+  u === ("READY" as SignatureRequestStatus);
+
+const readyStatus = new t.Type<
+  SignatureRequestStatus,
+  SignatureRequestStatus,
+  unknown
+>(
+  "status",
+  isReadyStatus,
+  (input, context) =>
+    isReadyStatus(input) ? t.success(input) : t.failure(input, context),
+  t.identity
+);
```

Comment:
this looks like overkill and there's some type mangling. an inline comparison is enough

Reply:
according to https://github.com/pagopa/io-functions-sign/pull/22#discussion_r968613274 , to use the validate function I must necessarily create a codec!

Reply:
you can use `t.literal("READY")` then